### PR TITLE
Fixes Wryn's tail and adds the ability to sting

### PR DESCRIPTION
**Pull request by Marginalorb**
## What Does This PR Do
This fixes an issue that Wryn have had for quite some time, and also gives them the ability to sting.
Stinging is a fluff ability that only does a few brute damage and costs somewhere in the ballpark of 20 stamina damage to the user as to prevent people from just spamming it on someone until they die a death of a thousand paper cuts.

A HUGE thank you to Kiyahitayika for figuring out how to, and coding, the ability for Wryn to sting.

## Why It's Good For The Game
This paves the way to potentially adding Wryn as a player species rather than letting them collect dust as a broken relic of the past.

## Images of changes
wryntail
wrynsting

## Changelog
:cl:
add: Adds the ability to sting to wryn, and a button to the HUD to do so
fix: Fixes broken tail code, allowing for wryn to have their bee tails again.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species/wryn.dm [hunk 1]
```diff
@@ -49,6 +49,106 @@
...
+
+//Define the Sting Action
+/datum/action/innate/wryn_sting
+	name = "Wryn Sting"
+	desc = "Readies Wryn Sting for stinging."
+	button_icon_state = "wryn_sting_off"		//Default Button State
+	check_flags = AB_CHECK_LYING | AB_CHECK_CONSCIOUS | AB_CHECK_STUNNED
+
+//What happens when you click the Button?
+/datum/action/innate/wryn_sting/Trigger()
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/species/wryn.dm:75]: Styling suggestion for this proc, use an early return format. Example:
```dm
/datum/action/innate/wryn_sting/Trigger()
	if(!..())
		return

	var/mob/living/carbon/user = owner

	if(user.restrained() || user.pulledby || user.buckled) //Is your Wryn restrained, pulled, or buckled? No stinging
		to_chat(user, "<span class='notice'>You need freedom of movement to sting someone!</span>")
		return

	if(user.wear_suit) //Is your Wryn wearing a Hardsuit or a Laboat that's blocking their Stinger?
		to_chat(user, "<span class='notice'>You must remove your hardsuit, labcoat, or jacket before using your Wryn stinger.</span>")
		return

	if(user.getStaminaLoss() <= 50)	//Does your Wryn have enough Stamina to sting?
		user.dna.species.button_on = TRUE
		UpdateButtonIcon()
		user.dna.species.select_target(user)
	else
		to_chat(user, "<span class='notice'>You feel too tired to use your Wryn Stinger at the moment.</span>")
		return
```

File: code/modules/mob/living/carbon/human/species/wryn.dm [hunk 2]
```diff
@@ -49,6 +49,106 @@
...
+	var/list/names = list()
+	for(var/mob/living/carbon/human/M in orange(1))
+		names += M
+	var/target = input("Select a Target: ", "Sting Target", null) as null|anything in names
+	if(!target)		//No one's around!
+		to_chat(user, "<span class='warning'>There's no one around to sting so you retract your stinger.</span>")
+		user.visible_message("<span class='warning'[user] retracts their stinger.</span>")
+		user.dna.species.button_on = FALSE
+		wryn_sting.UpdateButtonIcon()
+		return
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/species/wryn.dm:120]: Unnecessary return.
```suggestion

```

File: code/modules/mob/living/carbon/human/species/wryn.dm [hunk 3]
```diff
@@ -49,6 +49,106 @@
...
+	else			//Get ready, aim, fire!
+		user.visible_message("<span class='warning'> [user] prepares to use their Wryn stinger!</span>")
+		wryn_sting(user, target)
+	return
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/species/wryn.dm:124]:
```suggestion

```

File: code/modules/mob/living/carbon/human/species/wryn.dm [hunk 4]
```diff
@@ -49,6 +49,106 @@
...
+
+//What does the Wryn Sting do?
+/datum/species/wryn/wryn_sting(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	button_on = FALSE					//For when we Update the Button Icon
+	if(!(target in orange(1, user)))	//Dang, did they get away?
+		to_chat(user, "<span class='warning'>You are no longer adjacent to [target]. You retract your stinger for now.</span>")
+		user.visible_message("<span class='warning'[user] retracts their stinger.</span>")
+		wryn_sting.UpdateButtonIcon()
+		return
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/species/wryn.dm:133]:
```suggestion

```

File: code/modules/mob/living/carbon/human/species/wryn.dm [hunk 5]
```diff
@@ -49,6 +49,106 @@
...
+		target.apply_damage(dam, BRUTE, organ)
+		playsound(user.loc, 'sound/weapons/bladeslice.ogg', 50, 0)
+		add_attack_logs(user, target, "Stung by Wryn Stinger - [dam] Brute damage to [organ].")
+		if(target.restrained())			//Apply tiny BURN damage if target is restrained
+			if(prob(50))
+				user.apply_damage(2, BURN, target)
+				to_chat(target, "<span class='danger'>You feel a little burnt! Yowch!</span>")
+				user.visible_message("<span class='danger'>[user] is looking a little burnt!</span>")
+		wryn_sting.UpdateButtonIcon()
+		return
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/species/wryn.dm:149]:
```suggestion

```

File: code/modules/mob/living/carbon/human/species/_species.dm [hunk 6]
```diff
@@ -320,6 +323,14 @@
...
 	return
 
+// Wryn Sting - Wryn Sting
+/datum/species/proc/wryn_sting(mob/living/H, mob/living/M)
+	return
+
+// Wryn Sting - Select Target
+/datum/species/proc/select_target(mob/living/carbon/human/H)
+	return
+
```
[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/_species.dm:333]: I was trying to do this, but I wasn't sure how.
Would it be something like 
`/datum/species/wryn/proc/wryn_sting`
and
`/datum/species/wryn/proc/select_target`
or something else?

File: code/modules/mob/living/carbon/human/species/_species.dm [hunk 7]
```diff
@@ -165,6 +165,9 @@
 	// Species specific boxes
 	var/speciesbox
 
+	// Species specific vars
+	var/button_on = FALSE
+
```
[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/_species.dm:170]: One hundred percent agree! I struggled with doing that, though @__@. I'll ping you in Discord.

File: code/modules/mob/living/carbon/human/species/wryn.dm (hunk 1 above)
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/wryn.dm:75]: Buh. That makes it look so much nicer!!

File: code/modules/mob/living/carbon/human/species/_species.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/_species.dm]: Done!

File: code/modules/mob/living/carbon/human/species/_species.dm (hunk 7 above)
[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/_species.dm]: Thank you!

File: code/modules/mob/living/carbon/human/species/wryn.dm (hunk 1 above)
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/wryn.dm:75]: Did this and a little more. 
`if(user.getStaminaLoss() >= 50)	//Does your Wryn have enough Stamina to sting?
		to_chat(user, "<span class='notice'>You feel too tired to use your Wryn Stinger at the moment.</span>")
 else 
                user.dna.species.button_on = TRUE
		UpdateButtonIcon()
		user.dna.species.select_target(user)`

File: code/modules/mob/living/carbon/human/species/wryn.dm (hunk 5 above)
Comment (Kiyahitayika) [reply code/modules/mob/living/carbon/human/species/wryn.dm:145]: Wait, did I just add these back in?

**Comment by Kiyahitayika** [comment]
Realized we never posted footage of the stinger in action:
https://imgur.com/a/rt10AKH

**Comment by Necaladun** [comment]
Could probably do a bit more damage with the stamina cost tbh, it seems punching is just as good.

But balancing before everything else is ready is a bit silly. This is absolutely fine as a placeholder, once more work is done we can tweak all the numbers.

**Review by variableundefined — APPROVED** [review]
@Marginalorb 
Please deconflict

**Pull request merged**